src/components/Search.tsx

```diff
@@ -14,38 +14,31 @@ type SearchProps = {
   location: string;
   setLocation: (place: Geoname) => void;
 };
+
 export const Search = ({location, setLocation}: SearchProps) => {
   const geonames = useMemo(() => getGeonames(), []);
-  const map = geonames.reduce(
-    (acc: Record<string, Geoname>, val) => ((acc[val.name] = val), acc),
-    {},
-  );
-
-  const dataset = geonames.map((val, idx) => ({
-    id: idx.toString(),
-    title: val.name,
-  }));
+  const dataset = useMemo(() => getDataset(geonames), [])
 
   const handleSelect = (item: TAutocompleteDropdownItem) => {
     if (item && item.title !== null) {
-      setLocation(map[item.title]);
+      setLocation(geonames[item.title]);
     }
   };
 
   return (
     <GlassView containerStyle={styles.container} glassStyle={styles.glassCcontainer} blurStyle={{blurAmount: 25, blurType: 'light'}}>
       <AutocompleteDropdown
         clearOnFocus={false}
-        closeOnBlur={true}
+        closeOnBlur={false}
         closeOnSubmit={false}
         textInputProps={{ placeholder: 'Search location', placeholderTextColor: 'white', style: styles.textStyle }}
         onSelectItem={handleSelect}
-        dataSet={[{id: '9_000_000_000', title: location}, ...dataset]}
+        dataSet={[...dataset]}
         inputContainerStyle={styles.searchBar}
-        debounce={3000}
+        debounce={1500}
```

Comment:
why not 150 or even 50?

waiting 1500ms (or 3000ms) before updating would definetely make the search appear slow?

Reply:
The reason its there is to avoid starting multiple searches as the user types. Not sure how this works under the hood, but the performance could in theory get worse with a lower number I think?